public/video-ui/src/components/VideoUpload/Upload.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import {browserHistory} from 'react-router';
+import Icon from '../Icon';
+
+class VideoUpload extends React.Component {
+  getUploadPolicy = (event) => {
+    if(!this.props.video)
+      return;
+
+    if(event.target.files.length == 0) {
+      this.props.uploadActions.clearUploadPolicy();
+    } else {
+      this.props.uploadActions.getUploadPolicy(this.props.video.id, event.target.files[0]);
+    }
+  };
+
+  startUpload = () => {
+    if(!this.props.upload)
+      return;
+
+    this.props.uploadActions.startUpload(this.props.upload.policy, this.props.upload.file);
+  };
+
+  goToAtomPage = () => {
+    const atomId = this.props.video ? this.props.video.id : null;
+
+    if(atomId) {
+      this.props.uploadActions.clearUploadPolicy();
+      browserHistory.push(`/videos/${atomId}`);
+    }
+  };
+
+  header() {
```

Comment:
Also I think having all these small render functions actually makes it harder to read? I'd incorporate this into the main render method